lib/frontend.ml

```diff
@@ -163,7 +164,8 @@ module Make(C: S.CONFIGURATION with type 'a io = 'a Lwt.t) = struct
       let pages = Io_page.pages num in
       List.iter
         (fun (gref, page) ->
-           let id = gref mod (1 lsl 16) in
+           let id = nf.rx_id in
+           nf.rx_id <- (succ nf.rx_id) mod (1 lsl 16) ;
```

Comment:
I don't think it's certain that netback will use the pages in order (though it is likely, and mirage-net-xen's netback does this). It could, for example, take the page with ID=0 and start filling it from a slow device, then rapidly cycle through the other pages with a fast device. In that case, you might send another one with ID=0 while the first was being processed.